core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -602,8 +602,8 @@ public PlanOptimizers(
                 .addAll(projectionPushdownRules)
                 .add(new PushProjectionIntoTableScan(plannerContext, typeAnalyzer, scalarStatsCalculator))
                 .add(new RemoveRedundantIdentityProjections())
-                .add(new PushLimitIntoTableScan(metadata))
```

Comment:
The order of these rules doesn’t (shouldn’t) matter. They get applied iteratively until the plan can’t change any more.

Reply:
It shouldn't matter, but some connectors may want to only push-down limits when no predicates are present (eg: iceberg, or a theoretically improved hive connector implementation) and the previous order guarantees that the connector will see the scan without any predicates on the first invocation of the rule. In some situations, this could result in partition information being fetched before partition column predicates are attached.

Reply:
If there’s an intervening filter between a tablescan and a limit, this rule won’t be called, regardless of where it appears in this list.

Also, keep in mind that there’s no guarantee about the order in which the rules in a single IterativeOptimizer will be considered.